go/enclave/evm/evm_facade.go

```diff
@@ -104,23 +107,63 @@ func (exec *evmExecutor) execute(tx *common.L2PricedTransaction, from gethcommon
 		return nil, err
 	}
 
+	actualGasLimit := msg.GasLimit // before l1 cost is applied
```

Comment:
l2OnlyGasLimit

Reply:
It's the full gas limit - in the receipt we put full gas used.
adjustPublishing modifies it to l2 only gas limit, but the sanity check is we have leftover from the full thing to do the reads.